### PR TITLE
Fix(UG): Add section on interpreting errors

### LINKED ISSUE #337: UG: Update Command Parsing section to detail how to read errors and error behaviours

### PR DESCRIPTION
**Pull request by DrWala**
## What this does
This PR:
1. Closes #337 by adding a section on how to interpret and recover from errors

## How to test
1. cd to `docs/` folder of the project.
2. Run `bundle exec jekyll serve`.
3. Open the jekyll-hosted website in your browser (default address: http://127.0.0.1:4000).
4. Check that the UG has the changes listed above

### CODE REVIEW

**Review by stein414 — APPROVED** [review]
LGTM. Thanks for adding a detailed explanation on how to understand errors.

File: docs/UserGuide.md [hunk 1]
```diff
@@ -135,6 +135,62 @@ If `NAME_STRING` = `John p/ Tom`, the command keyed in could look like `radd p/[
...
+This error typically shows a message along the following lines: `Invalid command format! [Information on how to use
+the command]`
+
+![Invalid command format](images/error-invalid-cmd-format.png)
+
+Typically this happens because the command does not match the required format. Check if you are missing any parameters.
+A special case of this takes place when the `INDEX` value is not a number.
+
+##### Parameters do not conform to required values
+This error typically shows a message along the following lines: `[Parameter name] must be [some condition]`. E.g
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:158]: Should `E.g` be `E.g.`?

Saw this in other places. If you make the change, consider doing a search for `e.g`

File: docs/UserGuide.md [hunk 2]
```diff
@@ -135,6 +135,62 @@ If `NAME_STRING` = `John p/ Tom`, the command keyed in could look like `radd p/[
...
+
+![Invalid param format](images/error-invalid-param-format.png)
+
+Typically this happens because the parameter provided does not match the required format. Check if your parameter conforms
+to the format stated under the [command parameters](#command-parameters) section. The error message usually gives some
+helpful information in resolving the error too.
+
+##### Commands that require an `INDEX`
+The bounds of the `INDEX` parameter is documented [below](#index). This section will explain how errors with the `INDEX`
+will be dealt with
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:170]: Missing fullstop?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -135,6 +135,62 @@ If `NAME_STRING` = `John p/ Tom`, the command keyed in could look like `radd p/[
...
+  **only** an `INDEX` (be it valid or not) value is provided (e.g. `redit 1000` or `redit 1`): The app will inform you
+  that the format is invalid, and display the format that should be used. This is another specific case 
+  of [incorrect formats](#incorrect-formats), as the command compulsorily needs at least one of the optional parameters 
+  to be provided.
+  
+- If the `INDEX` is 0 or lesser (e.g `rdel -5`): The app will inform you that `INDEX` should be a positive integer.
+
+![Index non positive](images/error-idx-non-positive.png)
+
+- If the `INDEX` is above the range of the collection specified (e.g `rdel 100` but there are only 10 residents): The
```
Comment (benedictkhoomw) [inline docs/UserGuide.md:189]: > but there are only 10 residents

Maybe change this to

> but there are only 6 residents

to better match the image that follows.

**Review by benedictkhoomw — COMMENTED** [review]
Thank you for making this section. I think users will have a better idea of how to deal with errors after reading it. I left a few suggestions in comments for your consideration.

File: docs/UserGuide.md (hunk 1 above)
Comment (DrWala) [reply docs/UserGuide.md:158]: Done.

File: docs/UserGuide.md (hunk 2 above)
Comment (DrWala) [reply docs/UserGuide.md:170]: Done

File: docs/UserGuide.md (hunk 3 above)
Comment (DrWala) [reply docs/UserGuide.md:189]: Done

**Comment by DrWala** [comment]
All comments addressed.

**Pull request merged**